include/envoy/http/filter.h

```diff
@@ -222,6 +242,23 @@ class StreamDecoderFilterCallbacks : public virtual StreamFilterCallbacks {
    * It is not safe to call this from under the stack of a DownstreamWatermarkCallbacks callback.
    */
   virtual void removeDownstreamWatermarkCallbacks(DownstreamWatermarkCallbacks& callbacks) PURE;
+
+  /**
+   * This routine may be called to increase the buffer limit for decoder filters.
+   * If the limit set would lower the buffer limit it will be ignored.
```

Comment:
What's the rationale here for now allowing a lower limit? I think this is going to change the current behavior WRT to buffer filter. For example, in the current code, the defaults are I believe 256MiB stream limit (yes I agree this is too high). Someone that had previously setup buffer filter with a 5MiB limit I believe will now be getting a 256MiB limit by default, right? Can we just allow a filter to lower the limit also? Otherwise I'm not sure the best course of action here...

Reply:
My concern there was it makes it really hard to tell what the final limit is if one filter can raise it and another can lower it and it depends on the order they were installed.
This is another thing which kinda worked better with individual limits rather than the CM handling everything but I figured deterministic levels were better.  I could lower it and we could require people do math, we could allow raising or lowering but not both?

Reply:
What do you mean by "I could lower it and we could require people do math" ? FWIW I kind of think we should just allow filters to set the limit, whether higher or lower, and document as necessary. IMO this is not a super case, and the behavior would probably be pretty clear as long as we document filters that modify limits? (For example perhaps have an arch overview doc explaining some of this briefly, and in buffer filter doc, link back and explain what buffer filter does WRT to stream limits)?

Reply:
Ah, sorry, I meant if we ever have another filter say a DoS filter which lowers the limit, the net limit on a connection depends on the order of filter installation.  To me it feels overly complex to have to track down the order of things to calculate when you'll 413 but if you think it's cleaner it's certainly a simple enough change!

Reply:
How do you feel about lowering but not raising then?

Reply:
Better than mixing and matching.  It feels counter-intuitive to have a buffer filter which won't buffer as much as you want though :-P
It's the sort of thing I'd like config validation for - if one could run through all routes and complain about bad config it'd make me happier but I think that's non-trivial in the current incarnation.  Commenting it up is better than nothing and we can log(warn) if encountering it once we have rate limited log macros :-)

Reply:
Alright. I guess my vote is for now just allow to raise/lower, and put in a TODO to potentially add in config warnings? I think that's not *too* bad, since we could build a system during load that looks at various buffer settings and warns about oddities?

I do think it would be worth adding some docs in public docs at least to buffer filter on how that setting relates to other filter settings.